qcodes/dataset/param_spec.py

```diff
@@ -61,45 +64,79 @@ def depends_on(self):
 
     def add_inferred_from(
             self,
-            inferred_from: Sequence[Union['ParamSpec', str]]
-    ) -> None:
+            inferred_from: Sequence[Union['ParamSpec', str]]) -> None:
         """
         Args:
             inferred_from: the parameters that this parameter is inferred_from
         """
         self._inferred_from.extend(
             p.name if isinstance(p, ParamSpec) else p
-            for p in inferred_from
-        )
+            for p in inferred_from)
 
-    def add_depends_on(
-            self,
-            depends_on: Sequence[Union['ParamSpec', str]]
-    ) -> None:
+    def add_depends_on(self,
+                       depends_on: Sequence[Union['ParamSpec', str]]) -> None:
         """
         Args:
             depends_on: the parameters that this parameter depends on
         """
         self._depends_on.extend(
             p.name if isinstance(p, ParamSpec) else p
-            for p in depends_on
-        )
+            for p in depends_on)
 
     def copy(self) -> 'ParamSpec':
         """
         Make a copy of self
         """
-        return ParamSpec(
-            self.name,
-            self.type,
-            self.label,
-            self.unit,
-            self._inferred_from,
-            self._depends_on
-        )
+        return ParamSpec(self.name, self.type, self.label, self.unit,
+                         deepcopy(self._inferred_from),
+                         deepcopy(self._depends_on))
```

Comment:
oo, good catch. is it because these are lists, not tuples?

Reply:
Exactly, lists are mutable. I would also really like to get rid of `add_depends_on` and `add_inferred_from`, but I suspect that things in Delft would break then?

Reply:
true. please, keep them for now.